public/src/components/amounts/AmountsForm.tsx

```diff
@@ -0,0 +1,174 @@
+  },
+  rightCol: {
+    overflowY: 'auto',
+    flexGrow: 1,
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'flex-start',
+    paddingTop: spacing(6),
+    paddingLeft: spacing(6),
+  },
+}));
+
+const AmountsForm: React.FC<InnerProps<AmountsTests>> = ({
+  data: configuredAmounts,
+  setData: setConfiguredAmounts,
+  saveData: saveConfiguredAmounts,
+}: InnerProps<AmountsTests>) => {
+  const classes = useStyles();
+
+  const [selectedTest, setSelectedTest] = useState<AmountsTest | undefined>();
+
+  const onTargetSelected = (target: Territory) => {
+    const currentTest = configuredAmounts.find(test => test.target === target);
+    if (currentTest) {
+      setSelectedTest(currentTest);
+    } else {
+      setSelectedTest(undefined);
+    }
+  };
+
+  const createLocalTest = (selected: CountryOptions) => {
+    if (selected && selected.label && selected.code) {
+      const newTest = {
+        testName: `SUPPORTER_AMOUNTS_EVERGREEN__${selected.code}`,
+        liveTestName: `SUPPORTER_AMOUNTS_AB_TEST__${selected.code}`,
+        isLive: false,
+        target: selected.code,
+        // Need to calculate seed
+        seed: Math.floor(Math.random() * 1000000),
+        variants: [
+          {
+            variantName: 'CONTROL',
+            defaultContributionType: 'MONTHLY' as ContributionType,
+            displayContributionType: [
+              'ONE_OFF' as ContributionType,
+              'MONTHLY' as ContributionType,
+              'ANNUAL' as ContributionType,
+            ],
+            amountsCardData: {
+              ONE_OFF: {
+                amounts: [1],
+                defaultAmount: 1,
+                hideChooseYourAmount: false,
+              },
+              MONTHLY: {
+                amounts: [10],
+                defaultAmount: 10,
+                hideChooseYourAmount: false,
+              },
+              ANNUAL: {
+                amounts: [100],
+                defaultAmount: 100,
+                hideChooseYourAmount: false,
+              },
+            },
+          },
+        ],
+      };
+
+      const updatedTests = [...configuredAmounts, newTest];
+      setConfiguredAmounts(updatedTests);
+      setSelectedTest(newTest);
+    }
+  };
+
+  const updateLocalTest = (updated: AmountsTest) => {
+    const updatedTests = configuredAmounts.filter(t => t.testName !== updated.testName);
+    updatedTests.push(updated);
+    setConfiguredAmounts(updatedTests);
+  };
+
+  const deleteLocalTest = (test: AmountsTest) => {
+    const updatedTests = configuredAmounts.filter(t => t.testName !== test.testName);
+    setConfiguredAmounts(updatedTests);
+    setSelectedTest(undefined);
+  };
+
+  const saveLocalTestToS3 = () => {
+    saveConfiguredAmounts();
+  };
+
+  const getAllTestNames = () => {
+    const namesArray: string[] = [];
+    configuredAmounts.forEach(t => {
+      if (t.testName === selectedTest?.testName) {
+        namesArray.push(t.testName);
```

Comment:
This line is duplicated in the `if` and `else` blocks. Is that intentional? If it is perhaps it should be moved outside of the if/else?

Reply:
Also seems like a possible nice use case for `flatMap`?

```js
const getAllTestNames = () =>
  configuredAmounts.flapMap(t => {
    if (t.liveTestName) {
      return [t.testName, t.liveTestName];
    } else {
      return [t.testName];
    }
  });
```